hw/top_earlgrey/rdc/rdc_waivers.tcl

```diff
@@ -66,3 +66,47 @@ set_rule_status -rule E_RST_METASTABILITY -status Waived \
     (MetaStableFlop=~"*.u_rv_core_ibex.u_core.core_clock_gate_i.gen_generic.u_impl_generic.en_latch")} \
   -comment {If clockgating enble is reset, it affects the gating cell when \
     CLK is 0 value.}
+
+# PINMUX CSR to SPI_DEV PADs
+set_rule_status -rule E_RST_METASTABILITY -status Waived \
+  -expression {(ResetFlop=~"*.u_pinmux_aon.dio*.pull_en") && \
+    (MetaStableFlop=~"SPI_DEV_*")} \
+  -comment {POR_N resets SPI_DEV. The SPI_CLK portion is in idle state.}
+
+# PINMUX Retention to SPI_HOST
+set_rule_status -rule E_RST_METASTABILITY -status Waived \
+  -expression {(ResetFlop=~"*.u_pinmux_aon.dio_oe_retreg_q*") && \
+    (MetaStableFlop=~"SPI_HOST_CLK")} \
+  -comment {SPI_HOST_CLK is in idle when POR_N is asserted.}
+# PINMUX Retention to USB
+set_rule_status -rule E_RST_METASTABILITY -status Waived \
+  -expression {(ResetFlop=~"*.u_pinmux_aon.dio_oe_retreg_q*") && \
+    (MetaStableFlop=~"USB_*")} \
+  -comment {USB is in idle when POR_N is asserted.}
+
+# AES Clock Gating
+set_rule_status -rule E_RST_METASTABILITY -status Waived \
```

Comment:
this makes sense to me... but could these waivers in the future accidentally waive new resets we introduce only on the clkmgr side? Ie, something that resets the latch but now the consuming module? 

I just realized `POR_N` is not actually in the waiver itself here..so it would just look at the path. 
Is there anything we can do on the constraint or design side to make this better?

Reply:
is there a constraint that says something along the lines of..."everything" is reset..?

Reply:
Hum.. that's good point. I will split the glob into somewhat regex format. I don't know the tool supports it or not. If not, I will split it into each individual clock.

Reply:
o no sorry, i meant one of two things:

1. whether we could make the waiver specific to POR_N, so in the future if I do something stupid that added another reset that broke the rules, it wouldn't get waived. OR...

2.  Is there any way to specify the reset scenario that POR_N resets EVERYTHING, so don't worry about any of this assertion edge metastability..since no one will be alive.  This might be pretty similar to just waiving everything POR_N.  I was just curious if the tool had any semantic that allowed you to specify that.

Reply:
1. It is possible. Let me add this to POR_N case only

2. The tool analyzes it automatically. I suspect the reason why the tool reports this CG metastability issues is:

  a. The EN may create metastable clock for a cycle. The clock affects all the logic it connects to.
  b. The tool may be misconfigured to follow all the following logics (maybo max fanout or something like in cfg?).

I already asked AE for this issue and am waiting their response.

Reply:
ah i see.
Sounds good, thanks for explaining @eunchan 

This LGTM to me overall. If you're able to constrain 1 a bit that would be great, if not, maybe we just need to leave a comment that in the future, we should remove this waiver first before analyzing to make sure nothing is hidden.  Then add it back as appropriate.